src/Core/Domain/Product/Command/RemoveAllProductsFromPackCommand.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+
+/**
+ * Removes all products from provided pack
+ */
+class RemoveAllProductsFromPackCommand
+{
+    /**
+     * @var ProductId
+     */
+    private $packId;
+
+    /**
+     * @param int $packId
+     */
+    public function __construct(int $packId)
+    {
+        $this->packId = new ProductId($packId);
```

Comment:
Should it be renamed PackId? :thinking: 
It's confusing :sweat_smile: 
```suggestion
        $this->packId = new PackId($packId);
```

Reply:
I'm fifty/fifty between using ProductId or PackId. Both solutions provide benefits and cons.

I think @eternoendless insights will be helpful

Reply:
Same dilemma for me, since a Pack is also a Product, but it might indeed be confusing so it doesn't cost much to get rid of this ambiguity by introducing a `PackId` It would also encourage us to have stricter code (well we won't have the choice any more anyway 😅 )

Reply:
- Packs _are_ products, and they can be used interchangeably, so it would make sense to use ProductId
- Not all products are Packs, so it also makes sense to have a specific VO for it

Following the Liskov substitution principle, `PackId` is a subtype of `ProductId`

Reply:
SO we could just create a `PackId` class that extends `ProductId` without adding anything?

Reply:
Yes indeed 👍